providers/azure.go

```diff
@@ -126,6 +125,15 @@ func (p *AzureProvider) Redeem(ctx context.Context, redirectURL, code string) (*
 	if p.ProtectedResource != nil && p.ProtectedResource.String() != "" {
```

Comment:
Making a note of this here as well -- In the very related ADFS Provider PR, I was curious how feasible it was to use the default OIDC Provider or `oauth2.Config` for token redemption stuff. At least for ADFS, there's an opportunity to add this non-standard parameter into the `scope` parameter, is that a thing for Azure too?

https://github.com/oauth2-proxy/oauth2-proxy/pull/1021#discussion_r571505330